nifty/pipeline/nifsUtils.py

```diff
@@ -1163,3 +1185,47 @@ def MEFarith(MEF, image, op, result):
             iraf.imarith(operand1=result+'['+str(i)+']', op=op, operand2 = image, result = result+'['+str(i)+', overwrite]', divzero = 0.0)
 
 #-----------------------------------------------------------------------------#
+
+def download_query_cadc(program, directory='./rawData'):
+    """
+    Finds and downloads all CADC files for a particular gemini program ID to
+    the current working directory.
+    """
+
+    cadc = Cadc()
+    job = cadc.create_async("SELECT observationID, publisherID, productID FROM caom2.Observation \
+                             AS o JOIN caom2.Plane AS p ON o.obsID=p.obsID \
+                             WHERE instrument_name='NIFS' AND proposal_id={}".format("'"+program+"'"))
+    job.run().wait()
+    job.raise_if_error()
+    result = job.fetch_result().to_table()
+
+    # Store product id's for later
+    pids = list(result['productID'])
+
+    urls = cadc.get_data_urls(result)
+    for url, pid in zip(urls, pids):
+        try:
+            urllib.urlretrieve(url, directory+'/'+pid+'.fits')
```

Comment:
In stead of urllib can you use 'requests' .  That's a more abstracted interface that provides some extra robustness against changes in the lower library.  Also, this would allow you to look in the `header` of the HTTP `response` to see what the file name should be.  `requests` is part of the astroquery dependencies already.

Here is an example header: 
```Date: Thu, 07 May 2020 16:28:23 GMT
Strict-Transport-Security: max-age=0
Content-MD5: d13af3725015ebd2babd09a8f2eaac3d
ETag: d13af3725015ebd2babd09a8f2eaac3d
Content-Encoding: x-fits
Last-Modified: Mon, 04 Mar 2019 08:09:22 GMT
Content-Disposition: inline; filename=2376828o.fits.fz
X-Uncompressed-Length: 786038400
X-Uncompressed-MD5: 988909a91ab429bbb8a5f51b3d35715e
X-File-CRC: 
X-Uncompressed-CRC: 
Content-Type: application/fits
Content-Length: 330462720
Vary: Origin
Connection: close```
`Content-Disposition` can then be used to name the file.

Reply:
Good idea, see 69e03fd2 . Maybe this is what you're thinking.

Reply:
Yes, that looks better.   I made some comments in the commit.

Reply:
Perfect, thank you for all your feedback. Perhaps bbf9efe fixes this.